hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java

```diff
@@ -145,7 +145,8 @@ public class SerializationServiceV1 extends AbstractSerializationService {
     private static final int EE_BYTE_LENGTH = 2;
 
     private final PortableContextImpl portableContext;
-    private final PortableSerializer portableSerializer;
+    // Not final for testing
+    private PortableSerializer portableSerializer;
```

Comment:
And, of course, you have to use something like this to use this serialization service instead: https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/schema/CompactSchemaReplicationTestBase.java#L94

Reply:
We want to mock the portable serializer for sure because only one call to portableSerializer.readAsInternalGenericRecord comes from  `SerializationServiceV1.readAsInternalGenericRecord`. The others come from `PortableInternalGenericRecord.java` as `readAsInternalGenericRecord(BufferObjectDataInput in, int factoryId, int classId)`.

Regarding compact serializer, there is no problem because it was not final before.

Reply:
Then, what about returning a spied internal generic record here, and keeping track of readAsInternalGenericRecord calls over it?